### ISSUE TITLE
Post Process 3d Assets

### ISSUE THREAD

**Issue by ghost** [lucidrains/meshgpt-pytorch#4]
Hi, thank you for the wonderful code. I want to understand that, how can I post process the following as you mentioned in Usage. And, how do I run the code?

```
# after much training of transformer, you can now sample novel 3d assets

faces_coordinates = transformer.generate()

# (batch, num faces, vertices (3), coordinates (3))
# now post process for the generated 3d asset
```

**Comment by fire** [comment]
@lucidrains from https://github.com/lucidrains/meshgpt-pytorch/commit/9cb9614e761fdd0c5f97bebaa78fc9d87b60f7f3 you mentioned inputing these features.

I want to try running some samples in. Do you have any preference how to encode the samples from gltf models I have?

```csv
0x,0y,0z,1x,1y,1z,2x,2y,3z,area,0angle,1angle,2angle,0n,1n,2n
0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0
```

Here's one possible csv.

**Comment by fire** [comment]
So the mesh representation you want is

```
vertices = torch.randn((2, 121, 3))            # (batch, num vertices, coor (3))
faces = torch.randint(0, 121, (2, 64, 3))      # (batch, num faces, vertices (3))
```

Like

```
vertices = torch.tensor([
    [0.0, 0.0, 0.0],  # Vertex 1
    [1.0, 0.0, 0.0],  # Vertex 2
    [0.0, 1.0, 0.0],  # Vertex 3
    [0.0, 0.0, 1.0]   # Vertex 4
]).unsqueeze(0)       # Add batch dimension

faces = torch.tensor([
    [0, 1, 2],        # Face 1
    [0, 1, 3],        # Face 2
    [0, 2, 3],        # Face 3
    [1, 2, 3]         # Face 4
]).unsqueeze(0)       # Add batch dimension
```

**Comment by fire** [comment]
I'll put a pr with my code for review.

**Comment by MarcusLoppe** [comment]
I think there is some confusion. The output of the transformer is triangles and those contain vertexes.  
Regarding the angles,area and normal's, I've already explained in the code review but to repeat; they are automatically calculated so only the vertices and faces (indices/index to the vertices array) is needed. 

Some vertex can be duplicates since two triangles might use the same vertex, so you'll need to clean it up to be most memory effective.  
As per below, the triangles are a array with vertex, so 3 vertex and those vertex contains the positions XYZ (all though not sure about the order of the x y and z):

```[[-0.8047,  0.1953,  0.8516],
[-0.3359,  0.8203,  0.9609],
[ 0.3672, -0.5859,  `0.0078]]```

Output of 20 triangles:
```
torch.Size([1, 20, 3, 3])
tensor([[[[-0.8047,  0.1953,  0.8516],
          [-0.3359,  0.8203,  0.9609],
          [ 0.3672, -0.5859,  0.0078]],

         [[-0.4922, -0.7578, -0.3828],
          [-0.2422,  0.2266,  0.2578],
          [-0.2578, -0.8516, -0.3516]],
```

**Comment by fire** [comment]
I don't think we can know if the normals should be flat or smooth because the use case could be organic or mechanic. So if you autocalculate it would be either smooth or flat without a mix.
